feat(plugins-doc) update documentation for centreon-plugin-applications-protocol-[cifs,ftp,sftp] plugins (#3269)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-protocol-ftp.md

## Pack assets

### Templates

The Monitoring Connector **FTP Server** brings a host template:

* **App-Protocol-FTP-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="App-Protocol-FTP-custom" label="App-Protocol-FTP-custom">

| Service Alias | Service Template              | Service Description                                                |
|:--------------|:------------------------------|:-------------------------------------------------------------------|
| FTP-Login     | App-Protocol-FTP-Login-custom | Check connection on a remote FTP server with username and password |

> The services listed above are created automatically when the **App-Protocol-FTP-custom** host template is used.

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias  | Service Template                   | Service Description                                                              |
|:---------------|:-----------------------------------|:---------------------------------------------------------------------------------|

> | FTP-Commands   | App-Protocol-FTP-Commands-custom   | Check several commands execution on a remote FTP server                          |

Reviewer: ```suggestion
| FTP-Commands   | App-Protocol-FTP-Commands-custom   | Check  the execution of several commands on a remote FTP server                          |
```

| FTP-Date       | App-Protocol-FTP-Date-custom       | Check date of files within a directory or a specific file on a remote FTP Server |
| FTP-FilesCount | App-Protocol-FTP-FilesCount-custom | Count files on a remote FTP directory                                            |

> The services listed above are not created automatically when a host template is applied. To use them, [create a service manually](/docs/monitoring/basic-objects/services), then apply the service template you want.

</TabItem>
</Tabs>

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics linked to each service.

<Tabs groupId="sync">

> <TabItem value="FTP-Commands" label="FTP-Commands">
> 
> </TabItem>

Reviewer: ```suggestion
```

> <TabItem value="FTP-Commands" label="FTP-Commands">

Reviewer: there is a problem here

For this specific service template, the names of the metrics depend on the configuration made by the user.

</TabItem>
<TabItem value="FTP-Date" label="FTP-Date">

| Métrique   | Unité |
|:-----------|:------|
| files#list | N/A   |

</TabItem>
<TabItem value="FTP-FilesCount" label="FTP-FilesCount">

| Métrique    | Unité |
|:------------|:------|
| files       | N/A   |

</TabItem>
<TabItem value="FTP-Login" label="FTP-Login">

| Métrique | Unité |
|:---------|:------|
| time     | N/A   |

</TabItem>
</Tabs>

## Prerequisites

> User and password to connect to the ftp share.

Reviewer: ```suggestion
The plugin will need a username and password to connect to the FTP share.
```

```

</TabItem>
<TabItem value="Debian 11" label="Debian 11">

```bash
apt install centreon-plugin-applications-protocol-ftp
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-plugin-Applications-Protocol-Ftp
```

</TabItem>
</Tabs>

## Using the monitoring connector

### Using a host template provided by the connector

1. Log into Centreon and add a new host through **Configuration > Hosts**.
2. Fill in the **Name**, **Alias** & **IP Address/DNS** fields according to your resource's settings.
3. Apply the **App-Protocol-FTP-custom** template to the host. A list of macros appears. Macros allow you to define how the connector will connect to the resource, and to customize the connector's behavior.
4. Fill in the macros you want. Some macros are mandatory.

| Macro           | Description                                                                                                                              | Default value     | Mandatory   |
|:----------------|:-----------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | FTPUSERNAME     | Specify username for authentification.                                                                                                   |                   |             |

Reviewer: ```suggestion
| FTPUSERNAME     | Specify username for authentication.                                                                                                   |                   |             |
```

> | FTPPASSWORD     | Specify password for authentification.                                                                                                   |                   |             |

Reviewer: ```suggestion
| FTPPASSWORD     | Specify password for authentication.                                                                                                   |                   |             |
```

| WARNING      | Warning threshold in seconds.                                                                                                          |                   |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). |                   |             |

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
/usr/lib/centreon/plugins/centreon_protocol_ftp.pl \
	--plugin=apps::protocols::ftp::plugin \
	--mode=login \
	--hostname=10.0.0.1  \

> 	--username='' \
> 	--password='' \

Reviewer: ```suggestion
	--username='username' \
	--password=password' \
```

| --ftp-options | Add custom ftp options. Example: --ftp-options='Debug=1" --ftp-options='useSSL=1'                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |

> | --username    | Specify username for authentification                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |

Reviewer: ```suggestion
| --username    | Specify username for authentication                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
```

> | --password    | Specify password for authentification                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |

Reviewer: ```suggestion
| --password    | Specify password for authentication                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
```

</TabItem>
<TabItem value="FTP-Date" label="FTP-Date">

| Option        | Description                                                                         |
|:--------------|:------------------------------------------------------------------------------------|
| --hostname    | IP Addr/FQDN of the ftp host                                                        |
| --port        | Port used                                                                           |
| --ssl         | Use SSL connection Need Perl 'Net::FTPSSL' module                                   |
| --ftp-options | Add custom ftp options. Example: --ftp-options='Debug=1" --ftp-options='useSSL=1'   |

> | --username    | Specify username for authentification                                               |

Reviewer: ```suggestion
| --username    | Specify username for authentication                                               |
```

> | --password    | Specify password for authentification                                               |

Reviewer: ```suggestion
| --password    | Specify password for authentication                                               |
```

| --critical    | Critical threshold in seconds for each files (diff time)                            |
| --directory   | Check files in the directory (no recursive) (multiple option)                       |
| --file        | Check file (multiple option)                                                        |
| --timezone    | Set the timezone of display date. Can use format: 'Europe/London' or '+0100'.       |

</TabItem>
<TabItem value="FTP-FilesCount" label="FTP-FilesCount">

| Option        | Description                                                                         |
|:--------------|:------------------------------------------------------------------------------------|
| --hostname    | IP Addr/FQDN of the ftp host                                                        |
| --port        | Port used                                                                           |
| --ssl         | Use SSL connection Need Perl 'Net::FTPSSL' module                                   |
| --ftp-options | Add custom ftp options. Example: --ftp-options='Debug=1" --ftp-options='useSSL=1'   |

> | --username    | Specify username for authentification                                               |

Reviewer: ```suggestion
| --username    | Specify username for authentication                                               |
```

> | --password    | Specify password for authentification                                               |

Reviewer: ```suggestion
| --password    | Specify password for authentication                                               |
```

| --critical    | Critical threshold (number of files)                                                |
| --directory   | Check files in the directory (multiple option)                                      |
| --max-depth   | Don't check fewer levels (default: '0'. Means current dir only).                    |
| --filter-file | Filter files (can be a regexp. Directory is in the name).                           |

</TabItem>
<TabItem value="FTP-Login" label="FTP-Login">

| Option        | Description                                                                         |
|:--------------|:------------------------------------------------------------------------------------|
| --hostname    | IP Addr/FQDN of the ftp host                                                        |
| --port        | Port used                                                                           |
| --ssl         | Use SSL connection Need Perl 'Net::FTPSSL' module                                   |
| --ftp-options | Add custom ftp options. Example: --ftp-options='Debug=1" --ftp-options='useSSL=1'   |

> | --username    | Specify username for authentification                                               |

Reviewer: ```suggestion
| --username    | Specify username for authentication                                               |
```

> | --password    | Specify password for authentification                                               |

Reviewer: ```suggestion
| --password    | Specify password for authentication                                               |
```
